controllers/watchers/chaos_pod.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers
+
+import (
+	"fmt"
+
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	chaostypes "github.com/DataDog/chaos-controller/types"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/record"
+	k8scache "sigs.k8s.io/controller-runtime/pkg/cache"
+)
+
+// ChaosPodHandler handles the lifecycle of chaos pods to send notifications if needed
+type ChaosPodHandler struct {
+	// recorder is used to emit Kubernetes events.
+	recorder record.EventRecorder
+
+	// disruption is the current state of the disruption
+	disruption *chaosv1beta1.Disruption
+
+	// log is the logger.
+	log *zap.SugaredLogger
+}
+
+// NewChaosPodHandler creates a new instance of ChaosPodHandler
+func NewChaosPodHandler(r record.EventRecorder, d *chaosv1beta1.Disruption, l *zap.SugaredLogger) ChaosPodHandler {
+	return ChaosPodHandler{
+		recorder:   r,
+		disruption: d,
+		log:        l,
+	}
+}
+
+// OnAdd is a handler function for the add event of the controller.
+func (w ChaosPodHandler) OnAdd(_ interface{}) {
+	// Do nothing on add event
+}
+
+// OnUpdate is a handler function for the update event of the controller.
+func (w ChaosPodHandler) OnUpdate(oldObj, newObj interface{}) {
```

Comment:
This function is really simple. It sends notification if the pod is failing during the update.

The notification is not sent if: 
- The phase of the pod is not the same as before to avoid spamming.
- The previous phase of the pod is not in running.

If we want to catch all failures cases, we have to check the status of the chaos pod containers and its events and the events of the disruption to. It will complexify a lot the code. To know more you can refer to the [DisruptionTargetHanlder](https://github.com/DataDog/chaos-controller/pull/681/files#diff-ee1090926ca1d28cf941203c3d7e4cb7dcc035ff38a510f99b5b4876455d9699R60).